airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx

```diff
@@ -175,25 +183,34 @@ export const ConnectorCard: React.FC<ConnectorCardCreateProps | ConnectorCardEdi
         </div>
         {additionalSelectorComponent}
         <div>
-          <ConnectorForm
-            // Causes the whole ConnectorForm to be unmounted and a new instance mounted whenever the connector type changes.
-            // That way we carry less state around inside it, preventing any state from one connector type from affecting another
-            // connector type's form in any way.
-            key={selectedConnectorDefinition && Connector.id(selectedConnectorDefinition)}
-            {...props}
-            selectedConnectorDefinition={selectedConnectorDefinition}
-            selectedConnectorDefinitionSpecification={selectedConnectorDefinitionSpecification}
-            isTestConnectionInProgress={isTestConnectionInProgress}
-            onStopTesting={onStopTesting}
-            testConnector={testConnector}
-            onSubmit={onHandleSubmit}
-            formValues={formValues}
-            errorMessage={props.errorMessage || (error && generateMessageFromError(error))}
-            successMessage={
-              props.successMessage || (saved && props.isEditMode && <FormattedMessage id="form.changesSaved" />)
-            }
-            connectorId={isEditMode ? getConnectorId(props.connector) : undefined}
-          />
+          {props.isLoading && (
+            <div className={styles.loaderContainer}>
+              <Spinner />
+              <div className={styles.loadingMessage}>
+                <ShowLoadingMessage connector={selectedConnectorDefinition?.name} />
+              </div>
+            </div>
+          )}
+          {fetchingConnectorError && <FetchingConnectorError />}
+          {selectedConnectorDefinition && selectedConnectorDefinitionSpecification && (
```

Comment:
Is it possible that one of these is nullish even though a user has clicked on a connector from the dropdown? If so we may want to show some sort of error message here indicating that the connector could not be loaded or something. Because based on this code it still seems possible for `fetchingConnectorError` to be nullish, _and_ for one of these values to be nullish as well, in which case we just won't render anything here which feels like a not great UX

Reply:
I don't think there is a way to run into this situation besides the desired one when the user navigates to a completely empty form.

If you look at the code where the component is used in `SourceForm` and `DestinationForm`, the props `isLoading`,`fetchingConnectorError` and `selectedConnectorDefinition` are coming from a react-query hook, so one of them will always be set if there is a selection that has been made.